### ISSUE TITLE
Write a function for uploading new somas

### ISSUE THREAD

**Issue by jasper-tms** [htem/FANC_auto_recon#20]
### Current soma table organization
The latest (and hopefully final) soma table `somas_dec2022` contains both neuronal and glial nuclei. We then have reference tables (`simple_reference` schema) named `neuron_somas_dec2022` and `glia_somas_dec2022` that point to rows in `somas_dec2022` that are neurons or glia, respectively.

### Motivation
We occasionally discover cells whose nuclei weren't detected by the nucleus-detection CNN. We want to be able to upload new rows to the soma tables to indicate these.

### Implementation
It would be nice to have python functions named something like `upload_neuron_nucleus` and `upload_glia_nucleus` that will take care of first uploading a new row to `somas_dec2022`, then take care of uploading a new row to the relevant reference table. We can discuss specific argument options that we want in this thread.

@sumiya-kuroda are you OK implementing this sometime? At least implementing the basic functionality. I'm happy to contribute to adding additional options, but I haven't yet learned how to actually post rows to CAVE tables so having your help on at least that part would be appreciated.

**Comment by jasper-tms** [comment]
We also should discuss which module such a function should go into. We currently have `schema_upload` and `schema_download` modules but those are ancient and I think not much of the code there is still useful. I think a reasonable idea is to have a new `tables.py` module that contains functions for interacting with CAVE tables. The `upload_{neuron,glia}_nucleus` functions could go there. I'll be more sure after I have time to look through `schema_upload` and `schema_download` and decide what to keep vs trash.

**Comment by sumiya-kuroda** [comment]
I added new functions to upload new somas. Here is how you use them: you first need to initialize your CAVE client with `client = CAVEclient('fanc_production_mar2021')`. If you want to upload new somas of **neurons**, initialize `CAVEorganizer` like the codes below.
```
organizer = CAVEorganizer(client)
organizer.update_soma.initialize("somas_dec2022", "neuron")
```

`organizer.update_soma.soma_table` and `organizer.update_soma.subset_table` will give you up-to-date soma table and neuronal soma tables, respectively. `organizer.update_soma.preview()` will produce a Neuroglancer link with all the neuronal soma points available (but not working for now).

Then, you need to make `pandas.DataFrame` of somas that you want to add. Make sure it has the required columns. The data frame should look like the one in this picture.
<img width="195" alt="スクリーンショット 2022-12-16 17 47 07" src="https://user-images.githubusercontent.com/26479377/208174818-5a0cbe7e-a875-4dd5-b3a0-f77ce2fedd43.png">

If your data frame is `df`, their information will be uploaded to CAVE when you run this command. If everything works out, you will receive a message `Successfully uploaded!`
```
organizer.update_soma.add_dataframe(df)
```

Right now, **you have to wait ~1 hour after you have added new somas to the tables**. This is because the CAVE server looks up supervoxel IDs of newly added annotations every hour. The CAVE team is working on making this time window much shorter. Alternatively, we can "force" the CAVE server lookup svIDs of new somas using [ingest_annotation_table()](https://github.com/seung-lab/CAVEclient/blob/master/caveclient/materializationengine.py#L957), but this method sometimes doesn't work, and I am not using it for this version of the code.

**Comment by sumiya-kuroda** [comment]
For now I placed these functions in new Python scripts, but I'm happy to add them into some existing ones as well. `schema_upload` looks a little outdated and needs refinement, though.

**Comment by jasper-tms** [comment]
Thanks for doing this Sumiya! I agree that `schema_upload` is very much outdated at this point and not worth adding to. I suggest we have a new module for functions that help users upload things to cave. Maybe `upload.py`? If that sounds reasonable to you could you move your code (from both files you created) into a new file named `upload.py`?

In general I am opposed to having `utils` in module names – basically everything is a utility, so just leave it out so that users don't have to type `utils` in all their imports and function calls. I read this opinion somewhere recently and think it makes sense and so I'm tending to follow that guidance now. (You may notice that in the `reorganize` PR, I changed like 3 or 4 modules names to drop `utils`, e.g. `catmaid_utils.py` became `catmaid.py`, etc.)

**Comment by sumiya-kuroda** [comment]
Yes, `upload.py` sounds good to me. I will also delete soma-table-related functions that are not used anymore in `schema_upload.py` as well.

**Comment by sumiya-kuroda** [comment]
As we now have `nuclei_seg_Mar2022_verified` layer to store the nucleus segmentations for both automatically-found and manually-found somas in the soma table (see #27 ), we need to change the soma uploading functions in `upload.py` to use root IDs to annotate manually-found neurons. See [here](https://github.com/htem/FANC_auto_recon/issues/27#issuecomment-1374584131) for more discussion.

**Comment by jasper-tms** [comment]
Thanks @sumiya-kuroda, I've merged your improvements into `main` and added a few more commits myself. I was successfully able to upload a new soma using `add_soma`! I still have to write the function to transfer the nucleus segmentation to the `verified` layer, but after that we should be all set.

**Comment by jasper-tms** [comment]
Implemented in 8be9735605f1a9c5046f7384f972151b7323dca4. I've tested it a bit and it works for basic use, but I'm going to test it a bit more tomorrow before closing this issue.

It is super nice to be able to just call `fanc.upload.add_soma((x, y, z))` and have everything that needs to happen happen for the nucleus to be fully integrated. (Of course, sometimes it's necessary to call `fanc.upload.add_soma((x, y, z), nucleus_id=blah)` if there's not a good point to choose that's inside the nucleus segmentation object, but that's still quite easy to do.) Thanks for your work on this implementation Sumiya!
